src/axom/inlet/Table.cpp

```diff
@@ -0,0 +1,193 @@
+                                      const std::string& description)
+{
+  std::string fullName = getFullName(m_name, name);
+  axom::sidre::Group* sidreGroup = baseFieldAdd(fullName, description);
+  if (sidreGroup == nullptr)
+  {
+    //TODO: better idea?
+    return std::shared_ptr<Field>(nullptr);
+  }
+
+  bool value;
+  if(m_reader->getBool(fullName, value))
+  {
+    if (value)
+    {
+      sidreGroup->createViewScalar("value", (int8)1);
+    }
+    else
+    {
+      sidreGroup->createViewScalar("value", (int8)0);
+    }
+  }
+
+  return std::make_shared<Field>(sidreGroup);
+}
+
+std::shared_ptr<Field> Table::addDouble(const std::string& name,
+                                        const std::string& description)
+{
+  std::string fullName = getFullName(m_name, name);
+  axom::sidre::Group* sidreGroup = baseFieldAdd(fullName, description);
+  if (sidreGroup == nullptr)
+  {
+    return std::shared_ptr<Field>(nullptr);
+  }
+  
+  double value;
+  if(m_reader->getDouble(name, value))
+  {
+    sidreGroup->createViewScalar("value", value);
+  }
+
+  return std::make_shared<Field>(sidreGroup);
+}
+
+std::shared_ptr<Field> Table::addInt(const std::string& name,
+                                     const std::string& description)
+{
+  std::string fullName = getFullName(m_name, name);
+  axom::sidre::Group* sidreGroup = baseFieldAdd(fullName, description);
+  if (sidreGroup == nullptr)
+  {
+    return std::shared_ptr<Field>(nullptr);
+  }
+  
+  int value;
+  if(m_reader->getInt(name, value))
+  {
+    sidreGroup->createViewScalar("value", value);
+  }
+
+  return std::make_shared<axom::inlet::Field>(sidreGroup);
+}
+
+std::shared_ptr<Field> Table::addString(const std::string& name,
+                                        const std::string& description)
+{
+  std::string fullName = getFullName(m_name, name);
+  axom::sidre::Group* sidreGroup = baseFieldAdd(fullName, description);
+  if (sidreGroup == nullptr)
+  {
+    return std::shared_ptr<Field>(nullptr);
+  }
+  
+  std::string value;
+  if(m_reader->getString(name, value))
+  {
+    sidreGroup->createViewString("value", value);
+  }
+
+  return std::make_shared<Field>(sidreGroup);
+}
+
+std::shared_ptr<Table> Table::required(bool isRequired)
+{
+  SLIC_ASSERT_MSG(m_sidreGroup != nullptr, "Inlet's Table specific Sidre Datastore Group not set");
+
+  if (m_sidreGroup->hasView("required"))
+  {
+    std::string msg = fmt::format("Inlet Table has already defined required value: {0}",
+                                  m_sidreGroup->getName());
+    SLIC_WARNING(msg);
+    return shared_from_this();
+  }
+
+  if (isRequired)
+  {
+    m_sidreGroup->createViewScalar("required", (int8)1);
+  }
+  else
+  {
+    m_sidreGroup->createViewScalar("required", (int8)0);
+  }
+
+  return shared_from_this();
+}
+
+bool Table::required()
+{
+  SLIC_ASSERT_MSG(m_sidreGroup != nullptr, "Inlet's Table specific Sidre Datastore Group not set");
+
+  if (!m_sidreGroup->hasView("required"))
```

Comment:
We discussed this, but I think this would be a great use case for `sidre::Attributes`, which can have a default value of `false` (or 0, or whatever)